Urban-Fun/View Controllers/ActivityDetailsViewController.m

```diff
@@ -6,18 +6,90 @@
 //
 
 #import "ActivityDetailsViewController.h"
+#import "PFUser.h"
+#import "HelperClass.h"
 
 @interface ActivityDetailsViewController ()
 @property (strong, nonatomic) IBOutlet UILabel *detailsTitle;
+@property (strong, nonatomic) IBOutlet UILabel *detailsHostLabel;
+@property (strong, nonatomic) IBOutlet UILabel *detailsLocationLabel;
+@property (strong, nonatomic) IBOutlet UIButton *detailsFavoriteButton;
+@property (strong, nonatomic) IBOutlet UIButton *detailsJoinButton;
+@property (strong, nonatomic) IBOutlet UILabel *detailsAttendanceLabel;
+@property (strong, nonatomic) IBOutlet UILabel *detailsDescriptionLabel;
+@property (strong,nonatomic) PFUser *currentUser;
 
+- (void) setUpView;
 @end
 
 @implementation ActivityDetailsViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
+    [self setUpView];
+}
+
+- (IBAction)didTapJoin:(id)sender {
+    PFUser *currentUser = [PFUser currentUser];
+    if ([self.activity.host.objectId isEqualToString:currentUser.objectId]){
+        }
+    else{
+        [Activity updateAttendanceListWithUserId:currentUser.objectId withActivity:self.activity withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+            if ([self.activity.attendanceList containsObject:currentUser.objectId]){
+                [self.detailsJoinButton setSelected:YES];
+            }
+            else if (![self.activity.attendanceList containsObject:currentUser.objectId]){
+                [self.detailsJoinButton setSelected:NO];
+            }
+            [self setUpView];
+            
+        }];
+        
+    }
+    
+}
+
+- (void)setUpView {
     self.detailsTitle.text = self.activity.title;
+    self.detailsHostLabel.text = [NSString stringWithFormat:@"%@%@", @"Host - ", self.activity.host[@"username"]];
+    self.detailsAttendanceLabel.text = [NSString stringWithFormat:@"%@%lu", @"Attendance - ", (unsigned long)self.activity.attendanceList.count];
+    self.detailsDescriptionLabel.text = [NSString stringWithFormat:@"%@%@", @"Description - ", self.activity.activityDescription];
+    [self.detailsJoinButton setTitle:@"Join" forState:UIControlStateNormal];
+    PFUser *currentUser = [PFUser currentUser];
+    
+    if ([self.activity.host.objectId isEqualToString:currentUser.objectId]){
+        self.detailsLocationLabel.text = [NSString stringWithFormat:@"%@%@", @"Location - ", self.activity.address];
+    }
+    if ([_activity.attendanceList containsObject:currentUser.objectId]){
+        [self.detailsJoinButton setSelected:YES];
+        NSUInteger placeInLine = [_activity.attendanceList indexOfObject:currentUser.objectId];
```

Comment:
Curious, are we showing the address only to the people who join the event and is able to attend? if so, I think would be great to place a comment here since is a little bit confusing to me this logic.
I also suggest to split this logic into another method `setupLocationLabel` to handle this logic separately